src/threading.ts

```diff
@@ -117,12 +118,37 @@ class Thread {
     public writeTraceback(): void {
         this.traceback = "Traceback (most recent call last):\n" + this.traceback;
         process.stdout.write(this.traceback);
+        this.exit();
     }
 
-    // Output the traceback when thread cannot handle an exception, and exit the thread
-    public throwException(): void {
-        this.writeTraceback();
-        this.exit();
+    public throwException(exc: IPy_Object): void {
+        // Whenever an exception occurs, tries to find a handler and if it can't outputs the traceback 
+        // search in current frame
+        this.exc = exc;
+        var f: Py_FrameObject = <Py_FrameObject> this.getTopOfStack();
```

Comment:
Not all frames are `Py_FrameObjects`. Also, this code is not general for non-Python languages. We'll talk more in our meeting.